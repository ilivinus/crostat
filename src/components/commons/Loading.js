import React from 'react';
import { ActivityIndicator, View, StyleSheet } from 'react-native';

import { colors } from '../utils/constants';

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
});
//  size?: 'large' | 'small',


function Loading({ color = colors.primary, size = 'large' }){
  return (
    <View style={styles.root}>
      <ActivityIndicator color={color} size={size} />
    </View>
  );
}

export default Loading;