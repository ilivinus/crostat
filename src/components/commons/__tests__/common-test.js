import React from 'react'
import Loading from '../Loading';
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })

function setup() {
  const props = {
    color : '#fff',
    size : 23,
  }

  const enzymeWrapper = shallow(<Loading {...props} />)

  return {
    props,
    enzymeWrapper
  }
}
describe('#common components test',()=>{

    it('should',()=>{
        const {  enzymeWrapper, props } = setup();
        //expect(enzymeWrapper.find('ActivityIndicator'))
    })
    it('should render properly if passed nothing',()=>{
        const component = renderer.create(<Loading />).toJSON();

        expect(component).toMatchSnapshot();
    })
    it('should render white if pass color', () =>{
        const component = renderer.create(<Loading />).toJSON();

        expect(component).toMatchSnapshot();
    })

    it('should render large if pass size  large',()=>{
        const component = renderer.create(<Loading size='large'/>).toJSON();
        expect(component).toMatchSnapshot();
    })
})