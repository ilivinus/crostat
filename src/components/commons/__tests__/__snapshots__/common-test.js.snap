// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#common components test should render large if pass size  large 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#00C0FF"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`#common components test should render properly if passed nothing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#00C0FF"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`#common components test should render white if pass color 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#00C0FF"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
